docker/build_image.sh

```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+docker login -u $USERNAME --password $PASSWORD
+export TAG=`if [ "$TRAVIS_BRANCH" = "master" ]; then echo "latest"; else echo $TRAVIS_BRANCH ; fi`
+docker build --file=docker/wit.Dockerfile --tag=sifive/wit:$TAG .
+if [ "$TRAVIS_TAG" != "" ]; then docker push sifive/wit; fi
```

Comment:
It's a little bit misleading to have the script named `build_image.sh`, since it doesn't imply that an image would also get pushed, and one might attempt to run that without realizing that it will also cause a change in Docker Hub.

I think it would be a good idea to have the push go in a separate script. The `docker login` would also go with that, since it's only needed for pushing. You can have Travis only run that script as part of the `deploy:` top level key, which only runs when all the tests actually pass. You can also set rules so that the `deploy:` step only runs on certain branches/tags so that we don't attempt to do a push to Docker Hub on every single pull request.

See https://docs.travis-ci.com/user/deployment/script/ for more information.